### PR TITLE
Feature/show location and current tide status on widget (#28)

### PR DESCRIPTION
**Pull request by MarcoGuerrieriTAB**

### CODE REVIEW

File: TideApp/TideAppWidget/Scenes/TidesContentViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//  Created by Marco Guerrieri on 11/05/2021.
+//
+
+import Foundation
+
+struct TidesContentViewModel {
+  var place: String
+  var tideStatus: String?
+  var waterTemperature: String?
+  var tidesTimes: [WeatherData.Weather.Tide.TideData]?
```
[the remark this answers is not in the document]
Comment (MarcoGuerrieriTAB) [reply TideApp/TideAppWidget/Scenes/TidesContentViewModel.swift]: Done

File: TideApp/TideAppWidget/Domain/TidesWidgetDataProvider.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import Foundation
+import Combine
+import CoreLocation
+
+protocol TidesWidgetDataProviderType {
+  func retrieveData(completion: @escaping (TidesEntry.WidgetData) -> ())
+}
+
+class TidesWidgetDataProvider: TidesWidgetDataProviderType, ObservableObject {
```
[the remark this answers is not in the document]
Comment (MarcoGuerrieriTAB) [reply TideApp/TideAppWidget/Domain/TidesWidgetDataProvider.swift]: Done

File: TideApp/TideApp/Domain/Model/WeatherResponse.swift [hunk 3]
```diff
@@ -155,4 +135,28 @@ extension WeatherData {
...
+      return nil
+    }
+    let closestTides = date.closestDates(in: tideData.compactMap({ $0.tideDateTime.date(with: .dateTime)}))
+
+    guard let lastTideData = tideData.first(where: { data in
+      data.tideDateTime.date(with: .dateTime) == closestTides.first
+    }) else {
+      return nil
+    }
+    return lastTideData.tideType
```
[the remark this answers is not in the document]
Comment (MarcoGuerrieriTAB) [reply TideApp/TideApp/Domain/Model/WeatherResponse.swift]: Done

**Review by SophieClarkKC — APPROVED** [review]
🤩

**Review by jsanderson44 — APPROVED** [review]

**Pull request merged**